File: Geshotel/Geshotel.Web/Modules/Common/Imports/ServerTypings/Contratos.EdadesRow.ts

namespace Geshotel.Contratos {
    export interface EdadesRow {
        EdadesId?: number;
        HotelId?: number;
        ClienteId?: number;
        FechaDesde?: string;
        FechaHasta?: string;
        TipoHuespedId?: number;
        EdadMinima?: number;
        EdadMaxima?: number;
        UserId?: number;
        FechaModificacion?: string;
        HotelName?: string;
        Touroperador?: string;
        EmpresaId?: number;
        Empresa?: string;
        UserName?: string;
        TipoHuesped?: string;
    }

    export namespace EdadesRow {
        export const idProperty = 'EdadesId';
        export const nameProperty = 'Touroperador';
        export const localTextPrefix = 'Contratos.ContratosEdades';

        export namespace Fields {
            export declare const EdadesId: string;
            export declare const HotelId: string;
            export declare const ClienteId: string;
            export declare const FechaDesde: string;
            export declare const FechaHasta: string;
            export declare const TipoHuespedId: string;
            export declare const EdadMinima: string;
            export declare const EdadMaxima: string;
            export declare const UserId: string;
            export declare const FechaModificacion: string;
            export declare const HotelName: string;
            export declare const Touroperador: string;
            export declare const EmpresaId: string;
            export declare const Empresa: string;
            export declare const UserName: string;
            export declare const TipoHuesped: string;
        }

        [
            'EdadesId', 
            'HotelId', 
            'ClienteId', 
            'FechaDesde', 
            'FechaHasta', 
            'TipoHuespedId', 
            'EdadMinima', 
            'EdadMaxima', 
            'UserId', 
            'FechaModificacion', 
            'HotelName', 
            'Touroperador', 
            'EmpresaId', 
            'Empresa', 
            'UserName', 
            'TipoHuesped'
        ].forEach(x => (<any>Fields)[x] = x);
    }
}

File: Geshotel/Geshotel.Web/Modules/Contratos/ContratosEdades/ContratosEdadesRow.cs

namespace Geshotel.Contratos.Entities
{
    using Serenity;
    using Serenity.ComponentModel;
    using Serenity.Data;
    using Serenity.Data.Mapping;
    using System;
    using System.ComponentModel;
    using System.IO;

    [ConnectionKey("Default"), TableName("[dbo].[contratos_edades]"), DisplayName("Contratos Edades"), InstanceName("Contratos Edades"), TwoLevelCached]
    [ReadPermission("Contratos:General")]
    [ModifyPermission("Contratos:General")]
    public sealed class ContratosEdadesRow : Row, IIdRow
    {
        [DisplayName("Contratos Edades Id"), Column("contratos_edades_id"), Identity]
        public Int32? ContratosEdadesId
        {
            get { return Fields.ContratosEdadesId[this]; }
            set { Fields.ContratosEdadesId[this] = value; }
        }

        [DisplayName("Hotel Id"), Column("hotel_id"), NotNull]
        public Int16? HotelId
        {
            get { return Fields.HotelId[this]; }
            set { Fields.HotelId[this] = value; }
        }

        [DisplayName("Cliente Id"), Column("cliente_id"), NotNull]
        public Int32? ClienteId
        {
            get { return Fields.ClienteId[this]; }
            set { Fields.ClienteId[this] = value; }
        }

        [DisplayName("Fecha Desde"), Column("fecha_desde"), NotNull]
        public DateTime? FechaDesde
        {
            get { return Fields.FechaDesde[this]; }
            set { Fields.FechaDesde[this] = value; }
        }

        [DisplayName("Fecha Hasta"), Column("fecha_hasta"), NotNull]
        public DateTime? FechaHasta
        {
            get { return Fields.FechaHasta[this]; }
            set { Fields.FechaHasta[this] = value; }
        }

        [DisplayName("Tipo Huesped Id"), Column("tipo_huesped_id"), NotNull]
        public Int16? TipoHuespedId
        {
            get { return Fields.TipoHuespedId[this]; }
            set { Fields.TipoHuespedId[this] = value; }
        }

        [DisplayName("Edad Minima"), Column("edad_minima"), NotNull]
        public Int32? EdadMinima
        {
            get { return Fields.EdadMinima[this]; }
            set { Fields.EdadMinima[this] = value; }
        }

        [DisplayName("Edad Maxima"), Column("edad_maxima"), NotNull]
        public Int32? EdadMaxima
        {
            get { return Fields.EdadMaxima[this]; }
            set { Fields.EdadMaxima[this] = value; }
        }

        [DisplayName("User Id"), Column("user_id")]
        public Int32? UserId
        {
            get { return Fields.UserId[this]; }
            set { Fields.UserId[this] = value; }
        }

        [DisplayName("Fecha Modificacion"), Column("fecha_modificacion")]
        public DateTime? FechaModificacion
        {
            get { return Fields.FechaModificacion[this]; }
            set { Fields.FechaModificacion[this] = value; }
        }

        IIdField IIdRow.IdField
        {
            get { return Fields.ContratosEdadesId; }
        }

        public static readonly RowFields Fields = new RowFields().Init();

        public ContratosEdadesRow()
            : base(Fields)
        {
        }

        public class RowFields : RowFieldsBase
        {
            public Int32Field ContratosEdadesId;
            public Int16Field HotelId;
            public Int32Field ClienteId;
            public DateTimeField FechaDesde;
            public DateTimeField FechaHasta;
            public Int16Field TipoHuespedId;
            public Int32Field EdadMinima;
            public Int32Field EdadMaxima;
            public Int32Field UserId;
            public DateTimeField FechaModificacion;

            public RowFields()
                : base()
            {
                LocalTextPrefix = "Contratos.ContratosEdades";
            }
        }
    }
}
